queries/vhdl/highlights.scm

```diff
@@ -0,0 +1,407 @@
+  "for" @keyword)
+
+(component_configuration
+  "for" @keyword)
+
+(end_for
+  "for" @keyword)
+
+"return" @keyword.return
+
+[
+  "assert"
+  "report"
+  "severity"
+] @keyword.debug
+
+[
+  "if"
+  "then"
+  "elsif"
+  "case"
+] @keyword.conditional
+
+(when_element
+  "when" @keyword.conditional)
+
+(case_generate_alternative
+  "when" @keyword.conditional)
+
+(else_statement
+  "else" @keyword.conditional)
+
+(else_generate
+  "else" @keyword.conditional)
+
+[
+  "with"
+  "select"
+] @keyword.conditional.ternary
+
+(when_expression
+  "when" @keyword.conditional.ternary)
+
+(else_expression
+  "else" @keyword.conditional.ternary)
+
+(else_waveform
+  "else" @keyword.conditional.ternary)
+
+(else_expression_or_unaffected
+  "else" @keyword.conditional.ternary)
+
+"null" @constant.builtin
+
+(user_directive) @keyword.directive
+
+(protect_directive) @keyword.directive
+
+(warning_directive) @keyword.directive
+
+(error_directive) @keyword.directive
+
+(if_conditional_analysis
+  "if" @keyword.directive)
+
+(if_conditional_analysis
+  "then" @keyword.directive)
+
+(elsif_conditional_analysis
+  "elsif" @keyword.directive)
+
+(else_conditional_analysis
+  "else" @keyword.directive)
+
+(end_conditional_analysis
+  "end" @keyword.directive)
+
+(end_conditional_analysis
+  "if" @keyword.directive)
+
+(directive_body) @keyword.directive
+
+(directive_constant_builtin) @constant.macro
+
+(directive_error) @comment.error
+
+(directive_protect) @keyword.directive
+
+(directive_warning) @comment.warning
+
+[
+  (condition_conversion)
+  (unary_operator)
+  (logical_operator)
+  (relational_operator)
+  (shift_operator)
+  (sign)
+  (adding_operator)
+  (multiplying_operator)
+  (exponentiate)
+  (variable_assignment)
+  (signal_assignment)
+] @operator
+
+[
+  "'"
+  ","
+  "."
+  ";"
+] @punctuation.delimiter
+
+[
+  "("
+  ")"
+  "["
+  "]"
+  "<<"
+  ">>"
+] @punctuation.bracket
+
+[
+  ":"
+  "@"
+  "=>"
+] @punctuation.special
+
+[
+  (decimal_integer)
+  (string_literal_std_logic)
+] @number
+
+(decimal_float) @number.float
+
+(bit_string_length) @property
+
+(bit_string_base) @type.builtin
+
+(bit_string_value) @number
+
+(based_literal
+  (based_base) @type.builtin
+  (based_integer) @number)
+
+(based_literal
+  (based_base) @type.builtin
+  (based_float) @number.float)
+
+(string_literal) @string @spell
+
+(character_literal) @character
+
+(library_constant_std_logic) @constant.builtin
+
+(library_constant) @constant.builtin
+
+(library_function) @function.builtin
+
+(library_constant_boolean) @boolean
+
+(library_constant_character) @character
+
+(library_constant_debug) @keyword.debug
+
+(unit) @keyword.modifier
+
+(library_constant_unit) @keyword.modifier
+
+(label) @label
+
+(generic_map_aspect
+  "generic" @constructor
+  "map" @constructor)
+
+(port_map_aspect
+  "port" @constructor
+  "map" @constructor)
+
+(selection
+  (identifier) @variable.member)
+
+(_
+  view: (_) @type)
+
+(_
+  type: (_) @type)
+
+(_
+  library: (_) @module)
```

Comment:
Posibly - I'll experiment...

Reply:
This will work, but I'll get better behaviour by adding the "package" field to the subsequent identifiers (jpt13653903/tree-sitter-vhdl#27).  In other words, `use work.whatnot.all` will highlight `work` with `library: (_) @module` and `whatnot` with `package: (_) @module`.

I'm still a little hesitant though, because `use work.whatnot.func` will highlight `func` also with `package: (_) @module`, which I don't believe to be correct...  Maybe I must add the "package" field only to the first thing after the "library", and not any subsequent ones...  I'll experiment some more...

Reply:
I counsel against being too beholden to abstract "correctness", as this tends too lead to unusable highlighting queries due to their excessive complexity. (No comment on this particular case.)

Reply:
I have a solution ready and referenced in jpt13653903/tree-sitter-vhdl#27, just waiting on final comments from @ribru17